### PR TITLE
Pytorch backend for GLn and SPD matrices

### PR DESCRIPTION
**Pull request by ninamiolane**

### CODE REVIEW

File: geomstats/backend/pytorch.py [hunk 1]
```diff
@@ -10,6 +10,20 @@
 int32 = 'torch.LongTensor'
 int8 = 'torch.ByteTensor'
 
+CUDA = torch.cuda.is_available()
+DEVICE = torch.device("cuda" if CUDA else "cpu")
```
Comment (johmathe) [inline geomstats/backend/pytorch.py]: single quotes

File: geomstats/backend/pytorch.py [hunk 2]
```diff
@@ -317,6 +331,7 @@ def where(*args, **kwargs):
 
 def tile(x, y):
     # TODO(johmathe): Native tile implementation
+    y = tuple([int(one_y) for one_y in y])
```
Comment (johmathe) [inline geomstats/backend/pytorch.py]: - why need the tuple
- why need of the cast to an int?
- can't you do the cast to an int with something more backend friendly?


```
y.int() 
```

Comment (ninamiolane) [reply geomstats/backend/pytorch.py]: no need of the tuple -> corrected

need to cast to an int bcs gs.maximum is sometimes used as a index for tile and outputs a torch tensor (which is what we want it to do)
cannot do .int() bcs integers are also passed as indices

File: geomstats/backend/pytorch.py [hunk 3]
```diff
@@ -354,13 +373,6 @@ def diagonal(*args, **kwargs):
 
 def exp(*args, **kwargs):
     return torch.exp(*args, **kwargs)
```
Comment (johmathe) [inline geomstats/backend/pytorch.py]: why do we have 2 returns line by line?

**Pull request closed without merging**